python_modules/dagster/dagster/_check/__init__.py

```diff
@@ -167,13 +167,15 @@ def is_callable(obj: object, additional_message: Optional[str] = None) -> Callab
 # ##### CLASS
 # ########################
 
+T_Type = TypeVar("T_Type", bound=type)
+
 
 def class_param(
-    obj: object,
+    obj: T_Type,
```

Comment:
Not a huge fan of including this change until the discussion at https://github.com/dagster-io/internal/discussions/4618 has been resolved.